### ISSUE TITLE
pdoc3 does not generate html output

### ISSUE THREAD

**Issue by laimaretto** [pdoc3/pdoc#298]
### Expected Behavior
As per the information in [here](https://pdoc3.github.io/pdoc/doc/pdoc/#command-line-interface) I was expecting the documentation to be placed inside the `html` folder.

### Actual Behavior
Is not happening.

### Steps to Reproduce

1. `sudo pip3 install pdoc3`
2. `cd /myProject`
3. `pdoc3 --html -o ./html/ myProg.pu`

### Additional info

- pdoc version:
pip3 show pdoc3
Name: pdoc3
Version: 0.9.2
Summary: Auto-generate API documentation for Python projects.
Home-page: https://pdoc3.github.io/pdoc/
Author: UNKNOWN
Author-email: UNKNOWN
License: AGPL-3.0
Location: /usr/local/lib/python3.5/dist-packages
Requires: markdown, mako
Required-by:

**Comment by kernc** [comment]
What _is_ happening instead? Standard output, error?

**Comment by laimaretto** [comment]
Nothing. No error. My program just runs and that's it.

```
lucas@VMserver16:~/gen_config$ pdoc3 --html -o html/ get_config.py
/home/lucas/gen_config/config/out.cfg
lucas@VMserver16:~/gen_config$ 
lucas@VMserver16:~/gen_config$ ls html/ -l
total 0
lucas@VMserver16:~/gen_config$
```

**Comment by kernc** [comment]
Interesting. Instead, this should be happening:
https://github.com/pdoc3/pdoc/blob/49b7773ba6a96ddee63bf5c4cd48df250ab12c36/pdoc/cli.py#L532-L535
https://github.com/pdoc3/pdoc/blob/49b7773ba6a96ddee63bf5c4cd48df250ab12c36/pdoc/cli.py#L570-L573

What does this return?
```py
mod = pdoc.Module('get_config.py')
mod.html()
```
There's way too little for me to go by there (`cd yourProject` is not that reproducible), so welcome to investigate.

**Comment by laimaretto** [comment]
Ok, so if I run it that way, I do get an exception...

```
In [1]: import pdoc                                                                                                                                                                                                

In [2]: mod = pdoc.Module('get_config.py')                                                                                                                                                                         
/home/lucas/gen_config/config/out.cfg
An exception has occurred, use %tb to see the full traceback.

SystemExit: 0

/usr/local/lib/python3.5/dist-packages/IPython/core/interactiveshell.py:3334: UserWarning: To exit: use 'exit', 'quit', or Ctrl-D.
  warn("To exit: use 'exit', 'quit', or Ctrl-D.", stacklevel=1)
```
So I wonder if there are some requirements inside de `.py` for `pdoc` to work?

Thanks!

**Comment by kernc** [comment]
Maybe try absolute instead of relative path. Printed `/home/lucas/gen_config/config/out.cfg` rather indicates that _get_config.py_ is, in fact, imported and executed, is that correct?

**Comment by laimaretto** [comment]
So, I've tried as suggested the absolute path, not working either, both on the CLI or within the Python interpreter ... :-( 

1) CLI
```
lucas@VMserver16:~/gen_config$ pdoc3 --html -o html/ /home/lucas/gen_config/get_config.py
/home/lucas/gen_config/config/out.cfg
lucas@VMserver16:~/gen_config$ 
lucas@VMserver16:~/gen_config$ ls html/ -l
total 0
lucas@VMserver16:~/gen_config$ 
```
2) Interpreter
```
In [3]: import pdoc                                                                                                                                                                                                

In [4]: mod = pdoc.Module('/home/lucas/gen_config/get_config.py')                                                                                                                                    
/home/lucas/gen_config/config/out.cfg
An exception has occurred, use %tb to see the full traceback.

SystemExit: 0

```
---*---
Now, regarding your other question, `get_config.py` is just a little program that connects to a MySQL DB, gathers some info, and produces a configuration file `out.cfg` for a networking device. The path where the config file is located is shown at the end `/home/lucas/gen_config/config/out.cfg`. So, I believe the answer is no: `get_config.py` is not imported, I just execute it on the CLI.
```
lucas@VMserver16:~/gen_config$ python3 get_config.py 
/home/lucas/gen_config/config/out.cfg
```
What it is true, is that I do import some libraries from within `get_config.py`... would it a problem?

**Comment by kernc** [comment]
I don't know whether it's a problem. You should use a [main guard](https://stackoverflow.com/questions/19578308/what-is-the-benefit-of-using-main-method-in-python).

> ```
> In [4]: mod = pdoc.Module('/home/lucas/gen_config/get_config.py')                                                                                                                                    
> /home/lucas/gen_config/config/out.cfg
>
> SystemExit: 0
> ```
The fact that config line is printed out indicates the module is "run". That [`SystemExit`](https://docs.python.org/3/library/exceptions.html#SystemExit) is not pdoc's doing.

**Comment by laimaretto** [comment]
Indeed, now that you mention it, the `SystemExit: 0` is in my program.
```
if fnc_aux_test_config_file(var_config_file) == -1:
	# do some stuff
	sys.exit(1)
else:
	# Everything ok, writing files ...
	fnc_aux_write_config(var_config_file)
	sys.exit(0)
```
I do the previous because depending on the `sys.exit()` I can use that later on.

On the other hand, I don't know what you mean with 'the module is "run" ...'... :-P 

Finally, I'll have a look a the suggestion of the main guard: I don't have a main function in my code ... may be that will do it ... 

thanks !

**Comment by laimaretto** [comment]
Ok, I confirm that adding a main guard, at least produces an output. Thanks for the hint!

However, the render of the HTML file, seems a little incomplete. Let me share a snippet of my code so may be you'll understand.

1) I do some imports at the very beginning, with system imports and also some imports of my own... after that I have declared a function in the `get_config.py` file, which is my main file ...
```
import MySQLdb
import sys
import yaml
import pandas as pd

from templates.csr.csr import *

def fnc_sql_cfgTmplt(var_obj_id):
	"""Function that implements an SQL query towards DB"""
        return smthn.
```
2) The main guard: as you see, in the main section, I do call the previously declared function `fnc_sql_cfgTmplt()` but also a function called `fnc_sql_build_query1()` which is declared within my local imports `templates.csr.csr`.

```
if __name__ == '__main__':

	cfgTemplate = pd.read_sql(fnc_sql_cfgTmplt(var_obj_id), con = db)

	var_ports  = pd.read_sql(fnc_sql_build_query1(var_obj_id), con = db)
	db.close()
```
Well, in the produced HTML by `pdoc`, the only function that I can see is `fnc_sql_cfgTmplt()` which is the one declared inside `get_config.py`; none of the functions inside `templates.csr.csr` are shown ...

Any ideas? :-P ... thank you very much for your patience ... :-)

**Comment by kernc** [comment]
_Docs hierarchy follows module hierarchy_ is one of the imperatives. With `pdoc ... get_config.py` you're only telling it to document the one module. Point it at the parent package/directory to document all modules instead.

**Comment by laimaretto** [comment]
oh! I see ... brilliant ... :-) ... now it works ...`pdoc3 --html -o docs/doc.html . ` did the job ...

thank you very much ... !
